pkg/runtime/executor.go

```diff
@@ -278,11 +282,15 @@ func (e *Executor) findViolations(results *Output) error {
 }
 
 // implementsSubFolderScan checks if given iac type supports sub folder scanning
-func implementsSubFolderScan(iacType string) bool {
+func implementsSubFolderScan(iacType string, nonRecursive bool) bool {
 	// iac providers that support sub folder scanning
 	// this needs be updated when other iac providers implement
 	// sub folder scanning
-	iacWithSubFolderScan := []string{"all", "k8s", "helm"}
+	if nonRecursive && iacType == "terraform" {
+		return false
+	}
+
+	iacWithSubFolderScan := []string{"all", "k8s", "helm", "terraform"}
```

Comment:
Not in this PR, but we should have this information at the time of IaC registration and create a list of iac types which implement recursive scanning in the `init()` of package `iac-providers`. And we can fetch it from there, hence eliminating the need to hard code these values here.